spec/unit/relation/union_spec.rb

```diff
@@ -72,6 +72,94 @@
       end
     end
 
+    context "when the relations unioned have different names and virtual columns" do
+      let(:relation1) { relation.where(id: 1).select(:id, :name).select_append { [`NULL`.as(:title)] } }
+      let(:relation2) { tasks.select(:id).select_append { [`NULL`.as(:name)] }.select_append(:title) }
+
+      it "qualifies the table as the concatenated relation names" do
+        # this produces SQL like
+        # SELECT
+        #   `users__tasks`.`id`,
+        #   `users__tasks`.`name`,
+        #   NULL AS 'title'
```

Comment:
this is the problematic column select. if this got transformed to `users__tasks.title`, all would be well